CycloneDX/Services/GithubService.cs

```diff
@@ -0,0 +1,88 @@
+﻿// This file is part of the CycloneDX Tool for .NET
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright (c) Steve Springett. All Rights Reserved.
+
+using CycloneDX.Models;
+using Newtonsoft.Json;
+using System;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using License = CycloneDX.Models.License;
+
+namespace CycloneDX.Services
+{
+
+    public interface IGithubService
+    {
+        Task<License> GetLicenseAsync(string licenseUrl);
+    }
+
+    class GithubService : IGithubService
+    {
+
+        private string _baseUrl;
+        private HttpClient _httpClient;
+        private Regex _repositoryIdRegex = new Regex(@"https?\:\/\/[^\/]+\/(?<repositoryId>[^\/]+\/[^\/]+)\/.+$");
+
+        public GithubService(HttpClient httpClient, string baseUrl = null)
+        {
+            _httpClient = httpClient;
+            _baseUrl = baseUrl ?? "https://api.github.com/";
+        }
+
+        public async Task<License> GetLicenseAsync(string licenseUrl)
+        {
+            // Regex may let other URLs pass, a plain filter on URLs containing "github" may be enough
+            if (licenseUrl == null || !licenseUrl.Contains("github")) return null;
+
+            // Detect correct repository id starting from URL
+            var match = _repositoryIdRegex.Match(licenseUrl);
```

Comment:
@ST-Apps the regex to extract the repository id looks pretty relaxed. Is there a reason you haven't added the github check to the regex instead of the contains check?

Reply:
To be fair I was half asleep when I wrote it, sorry for that :)
I'm at work now and can't edit the code, but this one may be good enough to remove the `Contains`:

```
https?\:\/\/.*github.*[^\/]+\/(?<repositoryId>[^\/]+\/[^\/]+)\/.+$
```

I just did a quick test with [regex101](https://regex101.com/tests):

- https://github.com/CycloneDX/cyclonedx-dotnet/blob/master/LICENSE **matches** 
- https://raw.githubusercontent.com/CycloneDX/cyclonedx-dotnet/master/LICENSE **matches** 
- https://www.microsoft.com/web/webpi/eula/net_library_eula_enu.htm **doesn't match** 

If you think it's fine, I'll push again when I'll be back at home.

Reply:
Yeah, know what it's like to be coding late and wanting to get something finished off :)

I was thinking of being really specific. I think something like this would cover >90% of cases...

```
^https?\:\/\/((github\.com)|(raw\.githubusercontent\.com))\/(?<repositoryId>[^\/]+\/[^\/]+)\/blob\/master\/LICENSE$
```

This would also address the below issue as we would only ever be resolving the current licence if it was referenced specifically.

Reply:
Updated the regex starting from yours.
`blob` is not present on `raw.githubusercontent.com`, and `LICENSE` file can also have an extension so I allowed characters after `LICENSE`.